src/marqo/models/create_index_settings.py

```diff
@@ -36,26 +40,22 @@ class IndexSettings(StrictBaseModel):
     vectorNumericType: Optional[marqo_index.VectorNumericType] = Field(None, alias="vector_numeric_type")
     annParameters: Optional[marqo_index.AnnParameters] = Field(None, alias="ann_parameters")
 
-    @property
-    def request_body(self) -> dict:
-        return self.dict(exclude_none=True)
-
-
-class CreateIndexSettings(StrictBaseModel):
-    """A wrapper to create an index and a request body with explicit settings parameters
-    or with a settings dict"""
-    indexSettings: IndexSettings
-    settingsDict: Optional[Dict]
-
-    @property
-    def request_body(self) -> dict:
+    def generate_request_body(self) -> dict:
         """A json encoded string of the request body"""
-        return self.settingsDict if self.settingsDict is not None \
-            else self.indexSettings.request_body
+        # We return settingsDict if it is not None, otherwise we return the dict
+        if self.settingsDict is not None:
+            return self.settingsDict
+        else:
+            return self.dict(exclude_none=True, exclude={"settingsDict"})
 
     @root_validator(pre=True)
     def check_settings_dict_compatibility(cls, values):
-        if values.get("settings_dict") is not None and \
-                values.get("index_settings").dict(exclude_none=True) != {}:
-            raise ValueError("settings_dict and index_settings cannot be used together")
-        return values
+        """ Ensures that settingsDict is not specified along with other parameters.
+
+        Notes:
+            if you want to add a new parameter that can be specified along with settings_dict, add it to
+            SETTINGS_DICT_COMPATIBLE_PARAMS.
```

Comment:
does adding WAIT_FOR_READINESS to SETTINGS_DICT_COMPATIBLE_PARAMS solve the problem you mentioned?